vector/v.in.ogr/main.c

```diff
@@ -1175,13 +1181,23 @@ int main(int argc, char *argv[])
 			      Ogr_fieldname, OFTIntegerListlength);
 		}
 		else {
-		    G_warning(_("Column type (Ogr_ftype: %d) not supported (Ogr_fieldname: %s)"),
-			      Ogr_ftype, Ogr_fieldname);
-		    buf[0] = 0;
-		    col_info[i_out].type = G_store(buf);
-		}
-		G_free(Ogr_fieldname);
-	    }
+                /* handle columns of unsupported data type */
+                G_warning(_("Column type (Ogr_ftype: %d) not supported (Ogr_fieldname: %s)\n"),
+                          Ogr_ftype, Ogr_fieldname);
+                if (flag.drop->answer) {
+                    col_info[i_out] = col_info[i_out + 1];
+                    i_out--;
+                    ncols_out--;
+                    G_warning(_("Column <%s> will be omitted from import"),
+                              Ogr_fieldname);
```

Comment:
I'm not sure about the need to keep current broken behavior, but if the flag is explicitly used, then why warning? At max a verbose message, I would say, is what best reflects the user input.

The other G_warning is not needed (at least not as a warning) either for the same reason in case the flag is active.

Reply:
Now (in [47b4efd](https://github.com/OSGeo/grass/pull/2630/commits/47b4efd8760ea141183bbac639b79f0ec3ea3de0)) I prepared a solution with a new flag (`-d`) to drop unsupported columns from import. Without the `-d` -flag, import of a map with a binary column breaks the import.

I have no strong opinions about keeping the old behavior, yet, a break in the import may be a good thing, as it makes users aware of the issue. Also, other unsupported types that can be represented as text (not sure if there are (m)any though), may import good enough with the current behavior that redefines them as text. So columns are not dropped unless requested.

I removed the second warning.

Thanks, @nilason, @metzm and @wenzeslaus for the suggestions. Would be great if you could have a second look at the updated code...